internal/caller/linux/linuxconfig.go

```diff
@@ -0,0 +1,136 @@
+package caller
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+
+	"golang.zx2c4.com/wireguard/wgctrl/wgtypes"
+)
+
+var (
+	installPath   = "/etc/wireguard"
+	tunnelTemplate = `[Interface]
+PrivateKey = %s
+Address = %s
+	
+[Peer]
+PublicKey = %s
+AllowedIPs %s
+Endpoint = %s`
+)
+
+// LinuxConfig represents the tunnel configuration for a linux machine.
+type LinuxConfig struct {
+	localDevice wgtypes.Device
+	tunnel      string
+	confName    string
+	allowedIps  string
+	addr        string
+	endpoint    string
+}
+
+// init transmits the linux config factory to the caller Client.
+func init() {
+	AppConfigurator = NewConfig()
+}
+
+// NewConfig generates and returns a new LinuxConfig.
+func NewConfig() *LinuxConfig {
+	newDevice := wgtypes.Device{
+		Name: "Linux Configurator",
+		Type: wgtypes.Userspace,
+	}
+
+	newLinuxConfig := LinuxConfig{
+		localDevice: newDevice,
+		tunnel:      tunnelTemplate,
+		confName:    "",
+	}
+	return &newLinuxConfig
+}
+
+// GenerateKeys creates a new set of Private and Public Keys for the
+// LinuxConfig.
+func (l *LinuxConfig) GenerateKeys() error {
+	privKey, err := wgtypes.GeneratePrivateKey()
+	if err != nil {
+		return err
+	}
+	l.localDevice.PrivateKey = privKey
+	l.localDevice.PublicKey = privKey.PublicKey()
+	return nil
+}
+
+// UpdateTunnel refreshes the LinuxConfig's tunnel with the latest values.
+func (l *LinuxConfig) UpdateTunnel() {
+	newTunnel := fmt.Sprintf(tunnelTemplate,
+		l.localDevice.PrivateKey,
+		l.addr,
+		l.localDevice.PublicKey,
+		l.allowedIps,
+		l.endpoint)
+	l.tunnel = newTunnel
+}
+
+// InstallConfig installs the wireguard tunnel to the machine.
+func (l *LinuxConfig) InstallConfig() error {
```

Comment:
This is doing all kinds of wonky stuff with calling shell commands and using sudo.  It needs to be rewritten to not do any of that.  You probably will want to use renamio to atomically write the file, and some error handling to fail if the install directory isn't present (which you should probably take from an environment variable as well, since its not guaranteed to be in `/etc`).